ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -519,15 +543,16 @@ private void processReplicaMetadataResponse(Set<StoreKey> missingStoreKeys,
             "Blob id is not in the expected partition Actual partition " + blobId.getPartition()
                 + " Expected partition " + remoteReplicaInfo.getLocalReplicaId().getPartitionId());
       }
-      if (!missingStoreKeys.contains(messageInfo.getStoreKey())) {
+      BlobId localKey = (BlobId) remoteToLocal.get(messageInfo.getStoreKey());
+      if (!missingStoreKeys.contains(messageInfo.getStoreKey()) && localKey != null) {
```

Comment:
isn't it an exception if `localKey` is `null`? If it is, it should not be ignored silently since it indicates a bug

Reply:
added